addons/hr_holidays/models/hr_leave_type.py

```diff
@@ -116,8 +116,24 @@ def _search_valid(self, operator, value):
             or that don't need an allocation
             return [('id', domain_operator, [x['id'] for x in res])]
         """
-        date_to = self._context.get('default_date_from') or fields.Date.today().strftime('%Y-1-1')
-        date_from = self._context.get('default_date_to') or fields.Date.today().strftime('%Y-12-31')
+
+        if {'default_date_from', 'default_date_to', 'tz'} <= set(self._context):
+
+            default_date_from_str = self._context.get('default_date_from')  # sessions's local 07:00 given in UTC
+            default_date_to_str = self._context.get('default_date_to')  # sessions's local 19:00 given in UTC
+
+            # Convert:  "2024-07-23 05:00:00" -> datetime.datetime(2024, 7, 23, 5, 0)
```

Comment:
Same for this line, I feel like the variable names and the operation below are self-explanatory